tests/magnetostatics/test_semianalytic_2d.py

```diff
@@ -130,6 +130,20 @@ def semianalytic_results(zc):
             ax.set_ylabel("$B_{p}$ [T]")
             plt.show()
 
+    def test_tough_Bz_integration(self):
```

Comment:
What are we asserting in this test? Can we assert something about the return value?

If we only want to check that the function does not throw, then we should probably spell that out in the test name: `test_tough_Bz_integration_does_not_raise`